src/main/java/org/mule/extension/http/api/error/HttpError.java

```diff
@@ -32,6 +35,12 @@
 
   TIMEOUT,
 
+  // represents any 4xx error
+  CLIENT_ERROR((Predicate<Integer>) statusCode -> statusCode >= 400 && statusCode < 500),
```

Comment:
If an error handler matches `CLIENT_ERROR` and then `CLIENT_SECURITY` what would be the match for a 401 response?

Reply:
CLIENT_ERROR only will match if there aren't any specific error defined, eg
```java
      try {
        error = HttpError.valueOf(status.name());
      } catch (Throwable e) {
        error = stream(HttpError.values()).filter(httpError -> httpError.isParentOf(statusCode))
            .findFirst()
            .orElse(null);
      }
```
Because there is an UNAUTHORIZED error already defined, this would be the match.

Reply:
So it would go to `CLIENT_SECURITY`. I think this would work but we will need to be very careful documenting this. People might expect this to capture all client errors.

Reply:
@afelisatti, you are right.  We've discussed this problem with the team and decided that in order to fix it, we will just add the 5xx errors that were reported.  The more generic solution will be tackled in another JIRA.